File: ps/sum-of-left-leaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int sum;
    int sumOfLeftLeaves(TreeNode* root) {
        sum = 0;
        recurse(root);
        return sum;
    }
    void recurse(TreeNode* cur) {
        if (cur == NULL) return;
        if (cur->left != NULL){
            if ( cur->left->left == NULL && cur->left->right == NULL) {
                // cout << cur->val << endl;
                sum += cur->left->val;
            }
        } 
        recurse(cur->left);
        recurse(cur->right);
        return;
    }
};

File: ps/sum-of-left-leaves.py
# Definition for a binary tree node.
from typing import Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0
        if self.is_leaf(root.left):
            return root.left.val + self.sumOfLeftLeaves(root.right)
        if self.is_leaf(root.right):
            return self.sumOfLeftLeaves(root.left)

        return self.sumOfLeftLeaves(root.left) + self.sumOfLeftLeaves(root.right)

    def is_leaf(self, root: Optional[TreeNode]) -> bool:
        if root is None:
            return False
        if root.left is None and root.right is None:
            return True
        return False
